envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/JwtFilterFactory.kt

```diff
@@ -48,7 +48,9 @@ class JwtFilterFactory(
     }
 
     private fun shouldCreateFilter(group: Group): Boolean {
-        return group.proxySettings.incoming.endpoints.any { it.oauth != null } && properties.providers.isNotEmpty()
+        return group.proxySettings.incoming.endpoints.any {
+            it.oauth != null || it.clients.any { client -> client.selector in selectorToOAuthProvider.keys }
```

Comment:
Maybe: `selectorToOAuthProvider.keys.intersect(it.value).isNotEmpty()` instead of `it.clients.any { client -> client.selector in selectorToOAuthProvider.keys }`